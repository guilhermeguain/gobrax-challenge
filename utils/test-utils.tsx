import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react-native";
import type { RenderOptions } from "@testing-library/react-native";

import { Provider } from "react-redux";
import { AppStore, RootState, setupStore } from "@/store";

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
