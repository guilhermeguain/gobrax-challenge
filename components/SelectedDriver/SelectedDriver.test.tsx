import { screen } from "@testing-library/react-native";

import { renderWithProviders } from "@/utils/test-utils";

import { SelectedDriver } from "./";

describe("Component: SelectedDriver", () => {
  const name = "John Smith";
  const vehicle = { id: "1", brand: "Toyota", plate: "AAA-0000", driverId: "" };

  const mockup = {
    driver: {
      activeDriver: {
        id: "1",
        name,
        document: "000.000.000-00",
        vehicles: [vehicle],
      },
      drivers: [],
      hasError: false,
      isFetching: false,
    },
  };

  it("renders correctly", () => {
    const { toJSON } = renderWithProviders(<SelectedDriver />);

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly when driver's name is defined", () => {
    renderWithProviders(<SelectedDriver />, {
      preloadedState: mockup,
    });

    const element = screen.getByTestId("activeDriverName");

    expect(element).toBeTruthy();
    expect(element.props.children).toBe(name);
  });

  it("renders correctly when driver's name is undefined", () => {
    renderWithProviders(<SelectedDriver />);

    const element = screen.getByTestId("activeDriverName");

    expect(element).toBeTruthy();
    expect(element?.props.children).toBe(undefined);
  });

  it("renders correctly when driver's vehicle is defined", () => {
    renderWithProviders(<SelectedDriver />, {
      preloadedState: mockup,
    });

    const element = screen.getByTestId("activeDriverVehicle");

    expect(element).toBeTruthy();
    expect(element.props.children).toBe(
      `${vehicle?.brand} - ${vehicle?.plate}`
    );
  });

  it("renders correctly when driver's vehicle is undefined", () => {
    renderWithProviders(<SelectedDriver />);

    const element = screen.queryByTestId("activeDriverVehicle");

    expect(element).toBeFalsy();
  });
});
