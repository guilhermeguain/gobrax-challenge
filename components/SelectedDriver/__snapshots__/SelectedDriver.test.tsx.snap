// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SelectedDriver renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
    }
  }
>
  <View
    style={{}}
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <Text>
        Selecionado:
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Motorista:
      </Text>
      <Text
        testID="activeDriverName"
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        Ve√≠culo:
      </Text>
    </View>
  </View>
</View>
`;
